library/ssl_msg.c

```diff
@@ -724,32 +724,38 @@ int mbedtls_ssl_encrypt_buf( mbedtls_ssl_context *ssl,
                                     rec->data_len, 0 ) );
 
 #if defined(MBEDTLS_USE_PSA_CRYPTO)
-        status = psa_cipher_encrypt_setup( &cipher_op,
-                                     transform->psa_key_enc, transform->psa_alg );
+        /* Skip psa encryption for null cipher */
+        if ( transform->psa_alg != MBEDTLS_SSL_NULL_CIPHER )
```

Comment:
Hmm, sorry, I didn't think this through yesterday and earlier. I just realised I think this whole branch with `psa_cipher_encrypt` is dead code: there were only 2 stream ciphers in TLS, namely RC4 and NULL, and since we removed RC4 in 3.0, this means that the only way we have `mode == MBEDTLS_MODE_STREAM` and arrive here is if `psa_alg == MBEDTLS_SSL_NULL_CIPHER`.

So, I think this could be just

```
/* The only stream "cipher" we support is "NULL" */
if ( transform->psa_alg != MBEDTLS_SSL_NULL_CIPHER )
    return( MBEDTLS_ERR_SSL_INTERNAL_ERROR );

olen = rec->data_len;
```

Just out of precaution, perhaps don't remove the existing code yet: first get the CI to pass. Then do as I suggest above, and if I'm right the CI should still pass, otherwise I'll have learned something :)

Reply:
To be done when CI passes.